production/helm/loki/values.yaml

```diff
@@ -490,8 +490,9 @@ monitoring:
     namespace: null
     # -- Additional annotations for the dashboards ConfigMap
     annotations: {}
-    # -- Additional labels for the dashboards ConfigMap
-    labels: {}
+    # -- Labels for the dashboards ConfigMap
+    labels:
```

Comment:
what happens if someone provides custom labels? will it merge them or clobber this default? curious if instead we should just add this to the template. is this pattern documented anywhere?

Reply:
Dicts are merged, so that custom labels are simply added, and this one can only be overwritten.

I don't think it's a good idea to add this to the template, this is simply a de facto standard defined in the official grafana chart (right here https://github.com/grafana/helm-charts/blob/7f6a6634d3c095f0e448810be781bcfe5238f9d0/charts/grafana/values.yaml#L822 ) and as you can see, its value can be customized ("1" by default, although this is by incident :) https://github.com/grafana/helm-charts/issues/1457 ). Under the hood, a sidecar container (https://github.com/kiwigrid/k8s-sidecar) is used to grab configmaps by labels.

Anyway, this leaves the final user with following choices:
- don't touch anything in default loki and grafana charts values, and let magic happen 🪄 
- change default label value in grafana chart, and set the same value in loki chart
- change default label **name** in grafana chart, add the same to loki chart. The old one `grafana_dashboard` is still there, but who cares ? Imho, advantages outweigh the disadvantages :)